native/src/test/scala/sqs4s/internal/aws4/aws4suite/CanonicalRequestSpec.scala

```diff
@@ -0,0 +1,408 @@
+
+  "get vanilla empty query key" should "be transformed into canonical request correctly" in {
+    val req = Request[IO](
+      method = Method.GET,
+      uri = uri"/?Param1=value1",
+      headers = Headers.of(
+        Header("Host", "example.amazonaws.com"),
+        Header(XAmzDate, testTimeStamp)
+      )
+    )
+
+    CReq[IO](req).value.unsafeRunSync() shouldEqual expect(
+      "get-vanilla-empty-query-key"
+    )
+  }
+
+  "get vanilla query" should "be transformed into canonical request correctly" in {
+    val req = Request[IO](
+      method = Method.GET,
+      uri = uri"/",
+      headers = Headers.of(
+        Header("Host", "example.amazonaws.com"),
+        Header(XAmzDate, testTimeStamp)
+      )
+    )
+
+    CReq[IO](req).value.unsafeRunSync() shouldEqual expect("get-vanilla-query")
+  }
+
+  "get vanilla query order key" should "be transformed into canonical request correctly" in {
+    val req = Request[IO](
+      method = Method.GET,
+      uri = uri"/?Param1=value2&Param1=Value1",
+      headers = Headers.of(
+        Header("Host", "example.amazonaws.com"),
+        Header(XAmzDate, testTimeStamp)
+      )
+    )
+
+    CReq[IO](req).value.unsafeRunSync() shouldEqual expect(
+      "get-vanilla-query-order-key"
+    )
+  }
+
+  "get vanilla query order key case" should "be transformed into canonical request correctly" in {
+    val req = Request[IO](
+      method = Method.GET,
+      uri = uri"/?Param2=value2&Param1=value1",
+      headers = Headers.of(
+        Header("Host", "example.amazonaws.com"),
+        Header(XAmzDate, testTimeStamp)
+      )
+    )
+
+    CReq[IO](req).value.unsafeRunSync() shouldEqual expect(
+      "get-vanilla-query-order-key-case"
+    )
+  }
+
+  "get vanilla query order value" should "be transformed into canonical request correctly" in {
+    val req = Request[IO](
+      method = Method.GET,
+      uri = uri"/?Param1=value2&Param1=value1",
+      headers = Headers.of(
+        Header("Host", "example.amazonaws.com"),
+        Header(XAmzDate, testTimeStamp)
+      )
+    )
+
+    CReq[IO](req).value.unsafeRunSync() shouldEqual expect(
+      "get-vanilla-query-order-value"
+    )
+  }
+
+  "get vanilla query unreserved" should "be transformed into canonical request correctly" in {
+    val req = Request[IO](
+      method = Method.GET,
+      uri = canonicalUri[IO](
+        "/?-._~0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz=-._~0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz"
+      ).unsafeRunSync(),
+      headers = Headers.of(
+        Header("Host", "example.amazonaws.com"),
+        Header(XAmzDate, testTimeStamp)
+      )
+    )
+
+    CReq[IO](req).value.unsafeRunSync() shouldEqual expect(
+      "get-vanilla-query-unreserved"
+    )
+  }
+
+  "get vanilla utf8 query" should "be transformed into canonical request correctly" in {
+    val req = Request[IO](
+      method = Method.GET,
+      uri = canonicalUri[IO]("/?ሴ=bar").unsafeRunSync(),
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Non ascii characters are not allowed](https://app.codacy.com/manual/d2a4u/sqs4s/pullRequest?prid=4059007)